inference_server/model_handler/grpc_utils/generation_server.py

```diff
@@ -0,0 +1,53 @@
+import os
+from concurrent import futures
+
+import torch
+
+import grpc
+
+from ...constants import GRPC_MAX_MSG_SIZE
+from ...models import Model
+from ...utils import create_generate_request, print_rank_n
+from .pb import generation_pb2, generation_pb2_grpc
+
+
+class GenerationServer(generation_pb2_grpc.GenerationServiceServicer):
+    def __init__(self, model: Model) -> None:
+        self.model = model
+
+    def _unpack_proto_query_kwargs(self, query_kwargs):
+        query_kwargs = {k: getattr(v, v.WhichOneof("oneof_values")) for k, v in query_kwargs.items()}
+        return query_kwargs
+
+    def Generate(self, request, context):
+        text = [r for r in request.texts]
+        generate_kwargs = self._unpack_proto_query_kwargs(request.generate_kwargs)
+
+        request = create_generate_request(text=text, generate_kwargs=generate_kwargs)
+
+        local_rank = int(os.getenv("LOCAL_RANK", "0"))
+        torch.cuda.set_device(local_rank)
+        self.model.input_device = local_rank
+
+        response = self.model.generate(request)
+
+        response = generation_pb2.GenerationResponse(
+            texts=response.text, num_generated_tokens=response.num_generated_tokens
+        )
+        return response
+
+
+def serve(inference_pipeline, port):
+    server = grpc.server(
+        futures.ThreadPoolExecutor(max_workers=10),
```

Comment:
This should probably be `max_workers=1`? Since we don't want to serve parallel requests here

Reply:
Hmm, I saw MII was using 10 for some reason. But you are definitely right.